src/ConfigEditor.tsx

```diff
@@ -32,134 +29,96 @@ export function ConfigEditor(props: Props) {
   };
 
   // Catalogs
-  const [catalog, setCatalog] = useState<string | null>(jsonData.catalog || null);
   const fetchCatalogs = async () => {
     const res: string[] = await getBackendSrv().post(resourcesURL + '/catalogs', {
-      region: defaultKey,
+      region: props.options.jsonData.defaultRegion,
     });
     return res;
   };
   // Databases
-  const [database, setDatabase] = useState<string | null>(jsonData.database || null);
   const fetchDatabases = async () => {
     const loadedDatabases: string[] = await getBackendSrv().post(resourcesURL + '/databases', {
-      region: defaultKey,
-      catalog: catalog || defaultKey,
+      region: props.options.jsonData.defaultRegion,
+      catalog: props.options.jsonData.catalog,
     });
     return loadedDatabases;
   };
   // Workgroups
-  const [workgroup, setWorkgroup] = useState<string | null>(jsonData.workgroup || null);
   const fetchWorkgroups = async () => {
     const loadedWorkgroups: string[] = await getBackendSrv().post(resourcesURL + '/workgroups', {
-      region: defaultKey,
+      region: props.options.jsonData.defaultRegion,
     });
     return loadedWorkgroups;
   };
-  // OuputLocation
-  const [outputLocation, setOutputLocation] = useState<string | null>(jsonData.outputLocation || null);
 
   const onOptionsChange = (options: DataSourceSettings<AthenaDataSourceOptions, AthenaDataSourceSecureJsonData>) => {
-    // clean up related state
     setSaved(false);
-    setCatalog(null);
-    setDatabase(null);
-    setWorkgroup(null);
     props.onOptionsChange(options);
   };
 
-  const onCatalogChange = (catalog: string | null) => {
-    setCatalog(catalog);
+  const onChange = (resource: ResourceType) => (e: SelectableValue<string> | null) => {
+    const value = e?.value ?? '';
```

Comment:
i feel like how you have it is probably 100% but just for my own curiosity, whats the thinking in having a default be an empty string vs null or undefined or something like that?

Reply:
it's just about the type in `jsonData`. There it should be a string.